addons/mrp/models/mrp_workorder.py

```diff
@@ -231,32 +320,85 @@ def record_production(self):
         # Transfer quantities from temporary to final move line or make them final
         self._update_raw_moves()
 
+        # Record the final lot and quantity produced to suggest on next workorders
+        if self.product_tracking != 'none' and self.final_lot_id:
+            self._update_final_lot_line()
```

Comment:
_create_or_update_finished_line?
put this method next to _update_active_move_line